app/src/main/kotlin/org/fossify/filemanager/fragments/RecentsFragment.kt

```diff
@@ -178,9 +178,12 @@ class RecentsFragment(context: Context, attributeSet: AttributeSet) : MyViewPage
                         val name = cursor.getStringValue(FileColumns.DISPLAY_NAME) ?: path.getFilenameFromPath()
                         val size = cursor.getLongValue(FileColumns.SIZE)
                         val modified = cursor.getLongValue(FileColumns.DATE_MODIFIED) * 1000
-                        val fileDirItem = ListItem(path, name, false, 0, size, modified, false, false)
-                        if ((showHidden || !name.startsWith(".")) && activity?.getDoesFilePathExist(path) == true) {
+                        val isHiddenFile = name.startsWith(".")
+                        val isFileInHiddenFolder = isPathInHiddenFolder(path)
+                        val shouldShow = showHidden || (!isHiddenFile && !isFileInHiddenFolder)
```

Comment:
We can do the path check only when required to avoid unnecessary work:

```suggestion
                        val shouldShow = showHidden || (!isHiddenFile && !isPathInHiddenFolder(path))
```